test/dhcp/subnet_service_test.rb

```diff
@@ -0,0 +1,187 @@
+require 'test_helper'
+
+require 'dhcp/dhcp'
+require 'dhcp/subnet'
+require 'dhcp/record/lease'
+require 'dhcp/record/reservation'
+require 'dhcp/subnet_service'
+
+class SubnetServiceTest < Test::Unit::TestCase
+  def setup
+    @subnets = Proxy::MemoryStore.new
+    @leases_ip_store = Proxy::MemoryStore.new
+    @leases_mac_store = Proxy::MemoryStore.new
+    @reservations_ip_store = Proxy::MemoryStore.new
+    @reservations_mac_store = Proxy::MemoryStore.new
+    @reservations_name_store = Proxy::MemoryStore.new
+
+    @service = Proxy::DHCP::SubnetService.new(@subnets, @leases_ip_store, @leases_mac_store, @reservations_ip_store,
+                                              @reservations_mac_store, @reservations_name_store)
+  end
+
+  def test_add_subnet
+    subnet = Proxy::DHCP::Subnet.new("192.168.0.0", "255.255.255.0")
+    @service.add_subnet(subnet)
+
+    assert_equal 1, @service.all_subnets.size
+    assert @service.all_subnets.include?(subnet)
+  end
+
+  def test_should_not_add_duplicate_subnets
+    @service.add_subnet(Proxy::DHCP::Subnet.new("192.168.0.0", "255.255.255.0"))
+
+    assert_raise Proxy::DHCP::Error do
+      @service.add_subnet(Proxy::DHCP::Subnet.new("192.168.0.0", "255.255.255.0"))
+    end
+  end
+
+  def test_bulk_add_subnets
+    subnets = [Proxy::DHCP::Subnet.new("192.168.0.0", "255.255.255.0"),
+               Proxy::DHCP::Subnet.new("192.168.1.0", "255.255.255.0")]
+    @service.add_subnets(*subnets)
+
+    assert_equal 2, @service.all_subnets.size
+    assert @service.all_subnets.include?(subnets.first)
+    assert @service.all_subnets.include?(subnets.last)
+  end
+
+  def test_find_subnet
+    subnets = [Proxy::DHCP::Subnet.new("192.168.0.0", "255.255.255.0"),
+               Proxy::DHCP::Subnet.new("192.168.1.0", "255.255.255.0")]
+    @service.add_subnets(*subnets)
+
+    assert_equal subnets.first, @service.find_subnet("192.168.0.0")
```

Comment:
This should also be tested for the latter part of the method, to ensure it can find a subnet by an IP within the net, not just the network address.